modules/extending_and_customizing.md

```diff
@@ -0,0 +1,155 @@
+- [JHipster Kotlin](https://github.com/jhipster/jhipster-kotlin) blueprint replaces most of the server side Java code with equivalent Kotlin code.
+- [JHipster.NET](https://github.com/jhipster/jhipster-dotnetcore) blueprint replaces the entire server side with .NET implementation.
+- [JHipster NodeJS](https://github.com/jhipster/generator-jhipster-nodejs) blueprint replaces the entire server side with NestJS implementation.
+- [Svelte Hipster](https://github.com/jhipster/generator-jhipster-svelte) blueprint replaces the entire client side with Svelte implementation.
+- [JHipster Native](https://github.com/jhipster/generator-jhipster-native) blueprint customizes JHipster applications with Spring Native compatibility.
+
+### Side-by-side blueprint
+
+Each generator can be a sbs blueprint. A sbs blueprint doesn't change the original generator behavior, but can customize the behavior and the result.
+Side-by-side blueprint makes easier to support multiple JHipster versions and port to a new JHipster version.
+
+To make the generator side-by-side, add this to the constructor:
+
+```js
+this.sbsBlueprint = true;
+```
+
+Example: [server generator at Native Blueprint](https://github.com/mshima/generator-jhipster-native/blob/bb9c042f6bc70a26ba8037e951c93dc1d1820983/generators/server/generator.mjs#L17).
+At this example the generator [customizes package.json](https://github.com/mshima/generator-jhipster-native/blob/bb9c042f6bc70a26ba8037e951c93dc1d1820983/generators/server/generator.mjs#L26-L35), [removes files](https://github.com/mshima/generator-jhipster-native/blob/bb9c042f6bc70a26ba8037e951c93dc1d1820983/generators/server/generator.mjs#L37-L40), [customizes pom.xml](https://github.com/mshima/generator-jhipster-native/blob/bb9c042f6bc70a26ba8037e951c93dc1d1820983/generators/server/generator.mjs#L42-L186), [customizes java files](https://github.com/mshima/generator-jhipster-native/blob/bb9c042f6bc70a26ba8037e951c93dc1d1820983/generators/server/generator.mjs#L211-L307), [customizes cypress](https://github.com/mshima/generator-jhipster-native/blob/bb9c042f6bc70a26ba8037e951c93dc1d1820983/generators/server/generator.mjs#L321-L329) and so one.
+
+Side-by-side blueprint will be used to create hooks and help migrate a existing module and will be covered at [Creating a module](/modules/creating-a-module).
+
+### Custom cli
+
+Standalone blueprints can be executed using `yo`, but `yo` is aggressive in generators discovery (can be slow) and lacks some improvements JHipster cli provides like help and jhipster integration.
+So we recommend using `jhipster` cli or custom cli based on generator-jhipster.
+
+`jhipster` command executes the generator-jhipster version you have installed globally.
+A custom cli will execute the dependent generator-jhipster version, so will make sure to use the supported generator-jhipster version.
+
+Custom cli allows to execute a custom generator and will be covered at [Creating a module](/modules/creating-a-module).
+
+## Development
```

Comment:
A heading without anything under it seems weird. Maybe add a sentence here or collapse the heading below it.